mtp_noms_ops/apps/security/forms/object_list.py

```diff
@@ -588,3 +590,148 @@ def describe_field_amount_pattern(self):
             return _('ending in %02d pence') % amount_value
         description = dict(self['amount_pattern'].field.choices).get(value)
         return str(description).lower() if description else None
+
+
+@validate_range_fields(
+    ('triggered_at', _('Must be after the start date'), '__lt'),
+)
+class NotificationsForm(SecurityForm):
+    # NB: ensure that these templates are HTML-safe
+    filtered_description_template = 'All notifications are shown below.'
+    unfiltered_description_template = 'All notifications are shown below.'
+    description_templates = ()
+
+    page_size = 25
+
+    def __init__(self, request, **kwargs):
+        super().__init__(request, **kwargs)
+        self.date_count = 0
+
+    def get_object_list_endpoint_path(self):
+        return '/events/'
+
+    def get_query_data(self, allow_parameter_manipulation=True):
+        query_data = super().get_query_data(allow_parameter_manipulation=allow_parameter_manipulation)
+        if allow_parameter_manipulation:
+            query_data['rule'] = ('MONP', 'MONS')
+        return query_data
+
+    def get_api_request_page_params(self):
+        filters = super().get_api_request_page_params()
+        if filters is not None:
+            data = self.session.get('/events/pages/', params=filters).json()
+            self.date_count = data['count']
+            filters['ordering'] = '-triggered_at'
+            del filters['offset']
+            del filters['limit']
+            if data['newest']:
+                filters['triggered_at__lt'] = parse_date(data['newest']) + datetime.timedelta(days=1)
+                filters['triggered_at__gte'] = parse_date(data['oldest'])
+        return filters
+
+    def get_object_list(self):
+        events = convert_date_fields(super().get_object_list())
+        date_groups = map(summarise_date_group, group_events_by_date(events))
```

Comment:
Non-critical: I think nowadays python devs prefer list comprehension to map/filter as the former is clearer and more explicit.